### ISSUE TITLE
vs windows crash when updating a xml layout file (drawable)

### ISSUE THREAD

**Issue by softlion** [xamarin/xamarin-android#3106]
### Steps to Reproduce

1. Open an android project in VS2019 16.1.0 preview 3
2. Edit a text drawable

```xml
  <?xml version="1.0"?>
<description>System.AggregateException: One or more errors occurred. ---&gt; System.NullReferenceException: Object reference not set to an instance of an object.&#xD;
   at Xamarin.AndroidDesigner.Completion.AndroidLayoutCompletionEngine.get_CanUseJavaSession() in E:\A\_work\1622\s\Xamarin.Designer.Android\Xamarin.AndroidDesigner\Completion\AndroidLayoutCompletionEngine.cs:line 288&#xD;
   at Xamarin.AndroidDesigner.Completion.AndroidLayoutCompletionEngine.RenderXmlDrawable(String name, Int32 width, Int32 height, Boolean fromFrameworkResources) in E:\A\_work\1622\s\Xamarin.Designer.Android\Xamarin.AndroidDesigner\Completion\AndroidLayoutCompletionEngine.cs:line 292&#xD;
   at Xamarin.AndroidDesigner.Editor.QuickInfo.QuickInfoSource.&lt;RenderResourceValueAsync&gt;d__5.MoveNext() in E:\A\_work\1622\s\Xamarin.Designer.Android\Xamarin.AndroidDesigner\Editor\QuickInfo\QuickInfoSource.cs:line 125&#xD;
--- End of stack trace from previous location where exception was thrown ---&#xD;
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)&#xD;
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)&#xD;
   at Xamarin.AndroidDesigner.Editor.QuickInfo.QuickInfoSource.&lt;GetQuickInfoItemAsync&gt;d__4.MoveNext() in E:\A\_work\1622\s\Xamarin.Designer.Android\Xamarin.AndroidDesigner\Editor\QuickInfo\QuickInfoSource.cs:line 68&#xD;
--- End of stack trace from previous location where exception was thrown ---&#xD;
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)&#xD;
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)&#xD;
   at Microsoft.VisualStudio.Language.Intellisense.Implementation.AsyncQuickInfoSession.&lt;ComputeSourceContentAsync&gt;d__39.MoveNext()&#xD;
--- End of stack trace from previous location where exception was thrown ---&#xD;
   at Microsoft.VisualStudio.Telemetry.WindowsErrorReporting.WatsonReport.GetClrWatsonExceptionInfo(Exception exceptionObject)&#xD;
   --- End of inner exception stack trace ---&#xD;
   at Microsoft.VisualStudio.Language.Intellisense.Implementation.AsyncQuickInfoSession.&lt;UpdateAsync&gt;d__32.MoveNext()&#xD;
--- End of stack trace from previous location where exception was thrown ---&#xD;
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)&#xD;
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)&#xD;
   at Microsoft.VisualStudio.Language.Intellisense.Implementation.AsyncQuickInfoPresentationSession.&lt;UpdateAsync&gt;d__5.MoveNext()&#xD;
---&gt; (Inner Exception #0) System.NullReferenceException: Object reference not set to an instance of an object.&#xD;
   at Xamarin.AndroidDesigner.Completion.AndroidLayoutCompletionEngine.get_CanUseJavaSession() in E:\A\_work\1622\s\Xamarin.Designer.Android\Xamarin.AndroidDesigner\Completion\AndroidLayoutCompletionEngine.cs:line 288&#xD;
   at Xamarin.AndroidDesigner.Completion.AndroidLayoutCompletionEngine.RenderXmlDrawable(String name, Int32 width, Int32 height, Boolean fromFrameworkResources) in E:\A\_work\1622\s\Xamarin.Designer.Android\Xamarin.AndroidDesigner\Completion\AndroidLayoutCompletionEngine.cs:line 292&#xD;
   at Xamarin.AndroidDesigner.Editor.QuickInfo.QuickInfoSource.&lt;RenderResourceValueAsync&gt;d__5.MoveNext() in E:\A\_work\1622\s\Xamarin.Designer.Android\Xamarin.AndroidDesigner\Editor\QuickInfo\QuickInfoSource.cs:line 125&#xD;
--- End of stack trace from previous location where exception was thrown ---&#xD;
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)&#xD;
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)&#xD;
   at Xamarin.AndroidDesigner.Editor.QuickInfo.QuickInfoSource.&lt;GetQuickInfoItemAsync&gt;d__4.MoveNext() in E:\A\_work\1622\s\Xamarin.Designer.Android\Xamarin.AndroidDesigner\Editor\QuickInfo\QuickInfoSource.cs:line 68&#xD;
--- End of stack trace from previous location where exception was thrown ---&#xD;
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)&#xD;
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)&#xD;
   at Microsoft.VisualStudio.Language.Intellisense.Implementation.AsyncQuickInfoSession.&lt;ComputeSourceContentAsync&gt;d__39.MoveNext()&#xD;
--- End of stack trace from previous location where exception was thrown ---&#xD;
   at Microsoft.VisualStudio.Telemetry.WindowsErrorReporting.WatsonReport.GetClrWatsonExceptionInfo(Exception exceptionObject)&lt;---&#xD;
</description>
```
### Expected Behavior

No crash

### Actual Behavior

A pause in typing then a yellow bar appear "an error caused by an extension..."

### Version Information

[version.txt](https://github.com/xamarin/xamarin-android/files/3201510/Nouveau.document.texte.2.txt)

> VS bug [#897647](https://devdiv.visualstudio.com/DevDiv/_workitems/edit/897647)

**Comment by brendanzagaeski** [comment]
/cc @garuma for a quick first look when you get a chance. I searched a little bit for previous matching reports and didn't find anything right away, so I suspect this might require some slightly special circumstances to reproduce.

@softlion, as one little question in case it might come in handy as the team looks into this issue, for "text drawable," is that a `TextView` element? If it's easy to share, if you wouldn't mind pasting over the contents of a layout file that demonstrates the issue for you, that might be useful as the investigation progresses. Thanks!

**Comment by softlion** [comment]
Yes i can share it. It's so simple. But i don't know if it's caused by something else. It was one of these ones:


[shape_roundcorner_buttonplain.xml.txt](https://github.com/xamarin/xamarin-android/files/3210868/shape_roundcorner_buttonplain.xml.txt)
[shape_roundcorner_button_checked.xml.txt](https://github.com/xamarin/xamarin-android/files/3210872/shape_roundcorner_button_checked.xml.txt)
[shape_roundcorner_button_normal.xml.txt](https://github.com/xamarin/xamarin-android/files/3210873/shape_roundcorner_button_normal.xml.txt)

**Comment by xamarin-release-manager** [comment]
> [sync] [VS-25] Comment by **xamarinc**



Fixed in version 16.3.99.87 (master)



Author: Stephen Shaw  
 Commit: f00ac8679dbca31f0331a9832f5aaffc83765f64 (xamarin/designer)
